### PR TITLE
Improvements map interface

### PR DESCRIPTION
**Pull request by klemense1**
New features Map Interface:
- FindLane() (available through local map)
- hasCorrectDrivingDirection() (available through local map)
- isLineSegmentInsideDrivingCorridor() (used for RRT)

Cleaned up and restructured tests

### CODE REVIEW

File: modules/world/map/local_map.cpp [hunk 1]
```diff
@@ -28,6 +28,20 @@ LaneId LocalMap::GoalLaneIdFromGoalDefinitionPolygon(const GoalDefinitionPolygon
   return LaneId(0);
 }
 
+LanePtr LocalMap::FindLane(const Point2d& point) const {
+  LanePtr lane = map_interface_->FindLane(point);
+  return lane;
+}
+
+bool LocalMap::hasCorrectDrivingDirection(const State& state) const {
```
Comment (patrickhart) [inline modules/world/map/local_map.cpp]: Should be: HasCorrectDrivingDirection

File: modules/world/map/local_map.cpp [hunk 2]
```diff
@@ -28,6 +28,20 @@ LaneId LocalMap::GoalLaneIdFromGoalDefinitionPolygon(const GoalDefinitionPolygon
...
+  geometry::Point2d position(state(models::dynamic::StateDefinition::X_POSITION), state(models::dynamic::StateDefinition::Y_POSITION));
```
Comment (patrickhart) [inline modules/world/map/local_map.cpp]: Not longer than 80 lines; 

using models::dynamic::StateDefinition::X_POSITION;

above

File: modules/world/map/local_map.cpp [hunk 3]
```diff
@@ -28,6 +28,20 @@ LaneId LocalMap::GoalLaneIdFromGoalDefinitionPolygon(const GoalDefinitionPolygon
...
+  float orientation = state(models::dynamic::StateDefinition::THETA_POSITION);
+  std::cout << "orientation" << orientation << std::endl;
```
Comment (patrickhart) [inline modules/world/map/local_map.cpp]: Remove debug message

File: modules/world/map/map_interface.cpp [hunk 4]
```diff
@@ -105,7 +97,81 @@ bool MapInterface::FindNearestLanes(const Point2d &point,
   return true;
 }
 
-bool MapInterface::isInLane(const modules::geometry::Point2d &point, LaneId id) const
+LanePtr MapInterface::FindLane(const Point2d& point) const {
+  
+  LanePtr lane;
+  std::vector<LanePtr> nearest_lanes;
+  
+  // TODO: parameter (20) auslagern
```
Comment (patrickhart) [inline modules/world/map/map_interface.cpp]: add name for todo: // TODO(@esterle)

File: modules/world/map/map_interface.cpp [hunk 5]
```diff
@@ -105,7 +97,81 @@ bool MapInterface::FindNearestLanes(const Point2d &point,
...
+  if(!FindNearestLanes(point, 20, nearest_lanes, false)) {
```
Comment (patrickhart) [inline modules/world/map/map_interface.cpp]: Use ParameterServer straight away.. can define default value there

File: modules/world/map/map_interface.cpp [hunk 6]
```diff
@@ -105,7 +97,81 @@ bool MapInterface::FindNearestLanes(const Point2d &point,
...
+}
+
+bool MapInterface::hasCorrectDrivingDirection(const Point2d& point, const float orientation) const {
+
+  LanePtr lane = FindLane(point);
+  if (!lane) {
+    return false;
+  }
+
+  float orientation_normalized = remainder(orientation, 2*M_PI); // normalize to [0, 2pi]
```
Comment (patrickhart) [inline modules/world/map/map_interface.cpp]: two spaces before comment according to LINT

File: modules/world/map/map_interface.cpp [hunk 7]
```diff
@@ -105,7 +97,81 @@ bool MapInterface::FindNearestLanes(const Point2d &point,
...
+    if (same_orientation_as_road) {
+      return false;
+    }
+    else {
+      return true;
+    }
+  }
+}
+
+bool MapInterface::isLineSegmentInsideDrivingCorridor(const DrivingCorridorPtr corridor, const Point2d& p1, const Point2d& p2) const {
```
Comment (patrickhart) [inline modules/world/map/map_interface.cpp]: Shorten line

File: modules/world/map/map_interface.hpp [hunk 8]
```diff
@@ -42,14 +42,28 @@ class MapInterface {
...
+  bool FindNearestLanes(const Point2d& point,
                          const unsigned& num_lanes,
                          std::vector<opendrive::LanePtr>& lanes,
                          bool type_driving_only = true) const;
 
+  LanePtr FindLane(const Point2d& point) const;
+
+  bool hasCorrectDrivingDirection(const Point2d& point, const float orientation) const;
+
+  bool isLineSegmentInsideDrivingCorridor(const DrivingCorridorPtr corridor, const Point2d& p1, const Point2d& p2) const;
```
Comment (patrickhart) [inline modules/world/map/map_interface.hpp]: IsLineSegmentInsideDrivingCorridor

File: modules/world/map/map_interface.hpp [hunk 9]
```diff
@@ -42,14 +42,28 @@ class MapInterface {
...
+  * or inside the lane of any of them.
+  */
+  bool FindNearestLanes(const Point2d& point,
                          const unsigned& num_lanes,
                          std::vector<opendrive::LanePtr>& lanes,
                          bool type_driving_only = true) const;
 
+  LanePtr FindLane(const Point2d& point) const;
+
+  bool hasCorrectDrivingDirection(const Point2d& point, const float orientation) const;
```
Comment (patrickhart) [inline modules/world/map/map_interface.hpp]: HasCorrectDrivingDirection

File: modules/world/map/map_interface.hpp [hunk 10]
```diff
@@ -42,14 +42,28 @@ class MapInterface {
...
+
+  bool isLineSegmentInsideDrivingCorridor(const DrivingCorridorPtr corridor, const Point2d& p1, const Point2d& p2) const;
+
+  bool isInLane(const Point2d& point, LaneId id) const;
```
Comment (patrickhart) [inline modules/world/map/map_interface.hpp]: IsInLane

File: modules/world/tests/py_map_interface_tests.py [hunk 11]
```diff
@@ -113,10 +115,81 @@ def test_driving_corridor_splitting_4way_intersection(self):
...
+
+        map_interface.compute_all_driving_corridors()
+        all_corridors = map_interface.get_all_corridors()
+
+        point_sw = Point2d(46, 180)  # this point lies inside the sidewalk left of the road
+        point_rl = Point2d(52, 130)  # this point lies in the right lane
+        point_outside = Point2d(140, 37)  # this point lies far outside (in lane of type NONE)
+        point_no_lane = Point2d(120, 140)  # this point lies far outside, not in any lane
+
+        assert not map_interface.is_line_segment_inside_driving_corridor(all_corridors[0], point_sw, point_rl)
```
Comment (patrickhart) [inline modules/world/tests/py_map_interface_tests.py]: too long is_line_segment_inside_driving_corridor

File: modules/geometry/polygon.hpp [hunk 12]
```diff
@@ -154,6 +154,11 @@ inline bool Within(Point2d const& g1, Polygon const & g2) {
   return bg::within(g1, g2.obj_);
 }
 
+//! Line2d within polygon check, true if g1 is completely contained within g2, else false
+inline bool Within(const Line &l, const Polygon &poly) {
```
Comment (patrickhart) [inline modules/geometry/polygon.hpp]: Google style: Within(const Line& l, const Polygon& poly)

File: modules/world/map/map_interface.cpp (hunk 5 above)
Comment (klemense1) [reply modules/world/map/map_interface.cpp]: I've created an issue for that #156

**Pull request merged**